packages/test/src/workflows/updates-ordering.ts

```diff
@@ -0,0 +1,77 @@
+import { defineUpdate, setDefaultUpdateHandler, setHandler } from '@temporalio/workflow';
+
+const updateA = defineUpdate<ProcessedUpdate, [number]>('updateA');
+const updateB = defineUpdate<ProcessedUpdate, [number]>('updateB');
+const updateC = defineUpdate<ProcessedUpdate, [number]>('updateC');
+
+interface ProcessedUpdate {
+  handler: string;
+  updateName?: string;
+  args: unknown[];
+}
+
+/*
+    There's a surprising amount going on with the workflow below. Let's simplify it to just updateA and updateB 
+    (no updateC or the default) and walk through it.
+
+    1. setHandler for updateA
+    - this is all synchronous code until we run `UpdateScope.updateWithInfo(updateId, name, doUpdateImpl)`, 
+      which calls `doUpdateImpl` which is promise/async, so...
+    2. queue doUpdateImpl for A on node event queue: [doUpdateImplA]
```

Comment:
AFAICS, in the case of this specific workflow, `doUpdateImplA` will not get queued on node's event queue because there's no promise that is awaited from that update handler. Same for `updateB`.

Reply:
`doUpdateImplA` is async though, so I thought differently. Maybe we should coalesce this conversation into the other comment.